### PR TITLE
Multi-task BERT for NER

### LINKED ISSUE #78: Replace multi_task_lstm_crf load method with call to super
Replace 

```python
def load(self, weights_filepath, model_filepath):
        with open(model_filepath) as f:
            model = model_from_json(f.read(), custom_objects={'CRF': CRF})
            model.load_weights(weights_filepath)
            self.models.append(model)
```

with 

```python
def load(self, weights_filepath, model_filepath):
        super().load(weights_filepath, model_filepath, custom_objects={'CRF': CRF})
```

in `multi_task_lstm_crf.py`

### LINKED ISSUE #139: Multi-task BERT
Re-write the BERT model such that is can train on an arbitrary number of datasets, like the `MultiTaskLSTMCRF` model. A few things to keep in mind:

- For each dataset, the model should have its own output layer, optimizer compute its own loss function
- The interface should work the same as it does with the `MultiTaskLSTMCRF`, e.g.

```python
from saber import Saber

saber = Saber()
saber.load_dataset(['dataset_1', 'dataset_2'])

saber.build(model_name='bert-ner')
saber.train()
```

would load two datasets and create a multi-task model for training.

### PR DESCRIPTION
**Pull request by JohnGiorgi**
__Major changes__

This pull request replaces the BERT model for NER with a model that supports multi-task learning on an arbitrary number of input datasets. Like the `BiLSTMCRF` model, you simply need to load more than one dataset to use multi-task learning

```python
from saber import Saber

saber = Saber()
saber.load_dataset(['dataset_1', 'dataset_2'])

saber.build(model_name='bert-ner')
saber.train()
```

Unlike the `BiLSTMCRF`, this model implements smarter batching. During a training epoch, batches are randomly sampled from all the loaded datasets until each dataset is exhausted.

__Minor changes__

This pull request also introduces `prune_output_layers()`, 

```python
def prune_output_layers(self, indices):
    """Removes output layers with indicies not in `output_layer_indices` in `self.model`.

        Args:
            output_layer_indices (int or list): An integer index or list of inter indicies into the
                output layers of `self.model` to retain.

        Returns:
            `self.model`, where any output layers with indicies not in `output_layer_indices` have
            been removed.
        """
```

a method that can be called from any Saber model. This method accepts an index or list of indices corresponding to output layers to retain in a Saber model. All other output layers are dropped. E.g.

```python
saber.models[0].prune_output_layers([0, 2])
```

Would remove all output layers but the first and third from the saber model at `saber.models[0]`. This is useful when you train a model under the multi-task setting, but only want to retain one of the classifiers for inference/evaluation/deployment.

__Closes__

Closes #78. Closes #139.

### CODE REVIEW

File: saber/models/bert_for_token_classification_multi_task.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    token_type_ids = torch.LongTensor([[0, 0, 1], [0, 1, 0]])
+    config = BertConfig(vocab_size_or_config_json_file=32000, hidden_size=768,
+        num_hidden_layers=12, num_attention_heads=12, intermediate_size=3072)
+    num_labels = 2
+    model = BertForTokenClassification(config, num_labels)
+    logits = model(input_ids, token_type_ids, input_mask)
+    ```
+    """
+    def __init__(self, config, num_labels):
+        super(BertForTokenClassification, self).__init__(config)
```
[the remark this answers is not in the document]
Comment (JohnGiorgi) [reply saber/models/bert_for_token_classification_multi_task.py]: The fix that Codacy suggests breaks all my unit tests for `BertForNER`. Electing to ignore it for now.
